### PR TITLE
proposed task solution

### PR DESCRIPTION
**Pull request by wielkiborsuk**

### CODE REVIEW

File: src/components/Report/index.js [hunk 1]
```diff
@@ -1,8 +1,9 @@
 import React from 'react';
+import './index.css';
 
 const Report = props => {
     return (
-        <div className="single-raport">
+        <div className={"single-raport " + (props.active ? 'active-report' : '')}>
```
Comment (szymonsitko) [inline src/components/Report/index.js:6]: This is pure React & Redux way of styling components with conditional arguments. Good job. Also, if you would like to read more about it, please refer to this article:

https://medium.com/javascript-scene/10-tips-for-better-redux-architecture-69250425af44

and especially point number 5.

Also, using functional (stateless) component is better with Redux since it embraces better its paradigm.

File: src/reducers/Reports/index.js [hunk 2]
```diff
@@ -2,13 +2,16 @@ import * as types from '@constants/Types';
...
+    reportsList: [],
+    selected: null
 }
 
 export default (state = INITIAL_STATE, action) => {
     switch (action.type) {
         case types.TOGGLE_FETCHING_STATUS: return { ...state, fetchingReport: !state.fetchingReport }
         case types.SET_CURRENCIES_DATA: return { ...state, reportsList: state.reportsList.concat(action.payload) }
+        case types.SET_SELECTED_DATA: return { ...state, selected: action.payload }
+        case types.DELETE_SELECTED_DATA: return { ...state, reportsList: state.reportsList.filter((r) => r!==state.selected), selected: null }
```
Comment (szymonsitko) [inline src/reducers/Reports/index.js:14]: In order to make clear separation of concerns in your application, I do suggest making separate actions for filtering reportsList (as above) and separate one for clearing out selected element. Reducers with multiple state transformation are harder to debug and Redux gets little bit less modular.

My suggestion is to create:

Redux action:

const clearSelectedItem = () => { type: types.CLEAR_SELECTED_ITEM }

(no value passed, since it just assigns null to selected Redux store variable)

and corresponding reducer, like so:

...
case types.CLEAR_SELECTED_ITEM: return { ...state, selected: null }
...

PS. Props for filter, it's a common and good practice to use it like above.

**Review by szymonsitko — COMMENTED** [review]
In general I do like the concept of highlighting the current result which is controlled by Redux. If you are interested in that a littbe bit more, you can read more about Redux Saga, which also makes Redux more fun (also more complicated, since it uses generators, thus ES7 syntax).

Refer to:

https://github.com/redux-saga/redux-saga

File: src/reducers/Reports/index.js (hunk 2 above)
Comment (wielkiborsuk) [reply src/reducers/Reports/index.js:14]: Let me rephrase that, because it's not entirely clear to me.

You're not suggesting a whole new action/reducer section
Just a separation of:
- DELETE_SELECTED_DATA - that filters the list
- CLEAR_SELECTED_ITEM - that is responsible for changing the selected item in state

Am i right?

PS: If so, then i should probably let DELETE_SELECTED_DATA accept a parameter, to avoid issues with first clearing selection and then referencing it for deletion, right?
